code/datums/components/chem_booster.dm

```diff
@@ -175,12 +178,17 @@
 		return
 	wearer = equipper
 
+
 	configure_action.give_action(wearer)
 	power_action.give_action(wearer)
 	scan_action.give_action(wearer)
 	wearer.overlays += resource_overlay
 	update_resource(0)
 
+	RegisterSignal(wearer, COMSIG_KB_VALIHEAL, .proc/on_off)
```

Comment:
If a proc can be called by a signal, it should have SIGNAL_HANDLER on the first line of declaration. It should also not have any sleep in it(the signal handler thing is to check this), so you'll probably have to tweak some procs a bit to remove the sleep (add invoke_async probably)